core/templates/domain/exploration/StatesObjectFactorySpec.ts

```diff
@@ -336,56 +492,102 @@ describe('States object factory', () => {
       constants.NEW_STATE_TEMPLATE = oldNewStateTemplate;
     });
 
-    it('should create a new state given a state name', () => {
+    it('should create a new state given a state name and set ' +
+      'that state to a terminal state', () => {
       var newStates = ssof.createFromBackendDict(statesDict);
       newStates.addState('new state');
+
       expect(newStates.getState('new state')).toEqual(
-        sof.createFromBackendDict('new state', {
-          classifier_model_id: null,
+        sof.createFromBackendDict('new state', newState2));
+
+      expect(newStates.hasState('new state')).toBe(true);
+      expect(newStates.getStateNames()).toEqual(['first state', 'new state']);
+      expect((Object.keys(newStates.getStateObjects())).length).not.toBe(0);
+
+      newStates.setState('new state',
+        sof.createFromBackendDict('new state', newState));
+
+      expect(newStates.hasState('new state')).toBe(true);
+      expect(newStates.getStateNames()).toEqual(['first state', 'new state']);
```

Comment:
You're asserting these same two lines a few lines above. Can you assert if the newState was really set?

Reply:
Done